### ISSUE TITLE
Standard LB-enabled cluster fails to scale up

### ISSUE THREAD

**Issue by vossss** [Azure/aks-engine#2511]
**Describe the bug**
we are having issue when the aks engine cluster needs to scale up, upon creating new nodes , the autoscale status is on Failed

**Steps To Reproduce for Cluster-1**
On Cluster-1, the issue cannot be reproduced 100%, after several re-tries, the autoscale of the cluster was successful.

AKS Engine version
0.45.0 

Kubernetes version
v1.15.7

1. We created a cluster with this cluster definition:
`
{
  "apiVersion": "vlabs",
  "location": "chinanorth2",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.15",
      "orchestratorVersion": "1.15.7",
      "kubernetesConfig": {
        "loadBalancerSku": "Standard",
	"excludeMasterFromStandardLB": true,
	"kubeProxyMode" : "ipvs",
	"networkPlugin": "flannel",
        "privateCluster": {
          "enabled": true,
	  "jumpboxProfile": {
	     "name": "jumpbox",
	     "vmSize": "Standard_D2_v3",
	     "osDiskSizeGB": 30,
	     "username": "vmadmin",
	     "publickey": ""
	   }
        },
	"addons": [
        {
            "name": "cluster-autoscaler",
            "enabled": true,
            "pools": [
              {
                "name": "dpconnect",
                "config": {
                  "min-nodes": "5",
                  "max-nodes": "100"
                }
              }
            ],
            "config": {
              "scan-interval": "1m"
            }
          },
	  {
            "name": "heapster",
	    "enabled": true
          },
	  {
            "name": "dns-autoscaler",
            "enabled": false
          },
          {
            "name": "container-monitoring",
            "enabled": false
          },
          {
            "name": "tiller",
            "enabled": true,
	    "containers": [
              {
                "name": "tiller",
                "image": "gcr.azk8s.cn/kubernetes-helm/tiller:v2.13.0",
                "cpuRequests": "50m",
                "memoryRequests": "150Mi",
                "cpuLimits": "50m",
                "memoryLimits": "150Mi"
              }
            ],
            "config": {
              "max-history": "0"
            }
          },
          {
            "name": "blobfuse-flexvolume",
            "enabled": true
          },
          {
            "name": "keyvault-flexvolume",
            "enabled": true
          },
          {
            "name": "kubernetes-dashboard",
            "enabled": true
          },
          {
            "name": "aci-connector",
            "enabled": false
          },
          {
            "name": "smb-flexvolume",
            "enabled": false
          },
          {
            "name": "rescheduler",
            "enabled": false
          },
          {
            "name": "nvidia-device-plugin",
            "enabled": false
          }
        ]
       }
    },
      "masterProfile": {
        "count": 3,
        "dnsPrefix": "aksereh",
        "vmSize": "Standard_D2_v3",
        "vnetSubnetId": "/subscriptions/9e03d97f-ccca-428f-b508-ae42fd4b6af0/resourceGroups/rg-dp-connectivity-rehearsal/providers/Microsoft.Network/virtualNetworks/vnet-dp-connectivity-rehearsal/subnets/sbn-dp-connectivity-tier-10a-application-rehearsal",
        "firstConsecutiveStaticIP": "172.16.2.158",
        "distro": "aks-ubuntu-18.04"
      },
      "agentPoolProfiles": [
        {
            "name": "dpconnect",
            "count": 5,
            "vmSize": "Standard_D4s_v3",
            "vnetSubnetId": "",
            "distro": "aks-ubuntu-18.04",
            "availabilityProfile": "VirtualMachineScaleSets",
            "storageProfile": "ManagedDisks",
            "OSDiskSizeGB": 200            
        }
      ],
      "linuxProfile": {
        "adminUsername": "vmadmin",
        "ssh": {
          "publicKeys": [
            {
              "keyData": ""
            }
          ]
        }
      },
      "servicePrincipalProfile": {
        "clientId": "",
        "secret": ""
      },
      "certificateProfile": {}
  }
}
`
1.  The virtual machine scale set scaled up but the k8s-dpconnect-xxxx-vmss_x failed to upgrade. See linked images.
![图片](https://user-images.githubusercontent.com/16187999/71663684-1d1a6b80-2d91-11ea-82a5-799f93eb1831.png)
![图片](https://user-images.githubusercontent.com/16187999/71663716-3d4a2a80-2d91-11ea-8228-bc02a6c61894.png)
![图片](https://user-images.githubusercontent.com/16187999/71663731-476c2900-2d91-11ea-84ba-ed15310e02d9.png)
![图片](https://user-images.githubusercontent.com/16187999/71667888-e4cf5900-2da1-11ea-8de9-3e58cc483468.png)

**Steps To Reproduce for Cluster-2**
On Cluster-2, we used manual scale to sacle the cluster, but 100% failed.

AKS Engine version
0.45.0 

Kubernetes version
v1.15.7

1. changed instance count (17 -> 18)
![图片](https://user-images.githubusercontent.com/16187999/71668616-9cfe0100-2da4-11ea-8895-70361f1cbb80.png)
1. the instance can be created, but provisioning failed.
![图片](https://user-images.githubusercontent.com/16187999/71668828-62e12f00-2da5-11ea-8dd4-b4c909854842.png)
1. Like cluster-1 above, the reason is that the instance cannot connect to mirror.azk8s.cn.
![图片](https://user-images.githubusercontent.com/16187999/71669253-05e67880-2da7-11ea-9725-9005c827a0c0.png)

**Comment by CecileRobertMichon** [comment]
@vossss was the cluster created recently with 0.45.0 or was it an older cluster?

It looks like the reason it is failing is that CSE is unable to reach the mirror to download CNI... Is there an NSG rule blocking outbound internet? do the other VMs in the cluster have access to that mirror?

Also, 0.7.5 should already be on the VHD so it's strange that CSE is attempting to download it, can you please paste the content of `/opt/azure/vhd-install.complete`?

**Comment by vossss** [comment]
@CecileRobertMichon The clust was created recently with 0.45.0, 
The other VMs that are not scaled have access to that mirror.

the content of /opt/azure/vhd-install.complete:

root@k8s-dpcrb-52457148-vmss00000Y:~# cat /opt/azure/vhd-install.complete
Starting build on  Thu Dec 12 00:15:50 UTC 2019
Components downloaded in this VHD build (some of the below components might get deleted during cluster provisioning if they are not needed):
  - apache2-utils
  - apt-transport-https
  - auditd
  - blobfuse
  - ca-certificates
  - ceph-common
  - cgroup-lite
  - cifs-utils
  - conntrack
  - cracklib-runtime
  - ebtables
  - ethtool
  - fuse
  - git
  - glusterfs-client
  - init-system-helpers
  - iproute2
  - ipset
  - iptables
  - jq
  - libpam-pwquality
  - libpwquality-tools
  - mount
  - nfs-common
  - pigz socat
  - traceroute
  - util-linux
  - xz-utils
  - zip
  - etcd v3.3.15
  - moby v3.0.8
  - nvidia-docker2 nvidia-container-runtime
  - Azure CNI version 1.0.29
  - Azure CNI version 1.0.28
  - CNI plugin version 0.7.5
  - CNI plugin version 0.7.1
  - containerd version 1.2.4
  - containerd version 1.1.6
  - containerd version 1.1.5
  - img
Docker images pre-pulled:
  - k8s.gcr.io/kubernetes-dashboard-amd64:v1.10.1
  - k8s.gcr.io/exechealthz-amd64:1.2
  - k8s.gcr.io/addon-resizer:1.8.5
  - k8s.gcr.io/addon-resizer:1.8.4
  - k8s.gcr.io/addon-resizer:1.8.1
  - k8s.gcr.io/addon-resizer:1.7
  - k8s.gcr.io/heapster-amd64:v1.5.4
  - k8s.gcr.io/heapster-amd64:v1.5.3
  - k8s.gcr.io/heapster-amd64:v1.5.1
  - k8s.gcr.io/metrics-server-amd64:v0.3.5
  - k8s.gcr.io/metrics-server-amd64:v0.3.4
  - k8s.gcr.io/metrics-server-amd64:v0.2.1
  - k8s.gcr.io/k8s-dns-kube-dns-amd64:1.15.4
  - k8s.gcr.io/k8s-dns-kube-dns-amd64:1.15.0
  - k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.13
  - k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.5
  - k8s.gcr.io/kube-addon-manager-amd64:v9.0.2
  - k8s.gcr.io/kube-addon-manager-amd64:v9.0.1
  - k8s.gcr.io/kube-addon-manager-amd64:v9.0
  - k8s.gcr.io/kube-addon-manager-amd64:v8.9.1
  - k8s.gcr.io/kube-addon-manager-amd64:v8.9
  - k8s.gcr.io/kube-addon-manager-amd64:v8.8
  - k8s.gcr.io/kube-addon-manager-amd64:v8.7
  - k8s.gcr.io/kube-addon-manager-amd64:v8.6
  - k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64:1.15.4
  - k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64:1.15.0
  - k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64:1.14.10
  - k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64:1.14.8
  - k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64:1.14.5
  - mcr.microsoft.com/k8s/core/pause:1.2.0
  - k8s.gcr.io/pause-amd64:3.1
  - mcr.microsoft.com/k8s/azurestack/core/pause-amd64:3.1
  - gcr.io/kubernetes-helm/tiller:v2.13.1
  - gcr.io/kubernetes-helm/tiller:v2.11.0
  - gcr.io/kubernetes-helm/tiller:v2.8.1
  - k8s.gcr.io/cluster-autoscaler:v1.17.0
  - k8s.gcr.io/cluster-autoscaler:v1.16.2
  - k8s.gcr.io/cluster-autoscaler:v1.16.1
  - k8s.gcr.io/cluster-autoscaler:v1.16.0
  - k8s.gcr.io/cluster-autoscaler:v1.15.3
  - k8s.gcr.io/cluster-autoscaler:v1.15.2
  - k8s.gcr.io/cluster-autoscaler:v1.15.1
  - k8s.gcr.io/cluster-autoscaler:v1.15.0
  - k8s.gcr.io/cluster-autoscaler:v1.14.6
  - k8s.gcr.io/cluster-autoscaler:v1.14.5
  - k8s.gcr.io/cluster-autoscaler:v1.14.4
  - k8s.gcr.io/cluster-autoscaler:v1.14.2
  - k8s.gcr.io/cluster-autoscaler:v1.14.0
  - k8s.gcr.io/cluster-autoscaler:v1.13.7
  - k8s.gcr.io/cluster-autoscaler:v1.13.6
  - k8s.gcr.io/cluster-autoscaler:v1.13.4
  - k8s.gcr.io/cluster-autoscaler:v1.13.2
  - k8s.gcr.io/cluster-autoscaler:v1.13.1
  - k8s.gcr.io/cluster-autoscaler:v1.12.8
  - k8s.gcr.io/cluster-autoscaler:v1.12.7
  - k8s.gcr.io/cluster-autoscaler:v1.12.5
  - k8s.gcr.io/cluster-autoscaler:v1.12.3
  - k8s.gcr.io/cluster-autoscaler:v1.12.2
  - k8s.gcr.io/cluster-autoscaler:v1.3.9
  - k8s.gcr.io/cluster-autoscaler:v1.3.8
  - k8s.gcr.io/cluster-autoscaler:v1.3.7
  - k8s.gcr.io/cluster-autoscaler:v1.3.4
  - k8s.gcr.io/cluster-autoscaler:v1.3.3
  - k8s.gcr.io/cluster-autoscaler:v1.2.5
  - k8s.gcr.io/cluster-autoscaler:v1.2.2
  - k8s.gcr.io/k8s-dns-sidecar-amd64:1.14.10
  - k8s.gcr.io/k8s-dns-sidecar-amd64:1.14.8
  - k8s.gcr.io/coredns:1.6.5
  - k8s.gcr.io/coredns:1.6.2
  - k8s.gcr.io/coredns:1.5.0
  - k8s.gcr.io/coredns:1.3.1
  - k8s.gcr.io/coredns:1.2.6
  - k8s.gcr.io/coredns:1.2.2
  - k8s.gcr.io/rescheduler:v0.4.0
  - k8s.gcr.io/rescheduler:v0.3.1
  - microsoft/virtual-kubelet:latest
  - mcr.microsoft.com/containernetworking/networkmonitor:v0.0.6
  - mcr.microsoft.com/containernetworking/networkmonitor:v0.0.5
  - mcr.microsoft.com/containernetworking/azure-npm:v1.0.29
  - mcr.microsoft.com/containernetworking/azure-npm:v1.0.28
  - mcr.microsoft.com/containernetworking/azure-vnet-telemetry:v1.0.29
  - mcr.microsoft.com/containernetworking/azure-vnet-telemetry:v1.0.28
  - nvidia/k8s-device-plugin:1.11
  - nvidia/k8s-device-plugin:1.10
  - docker.io/deis/hcp-tunnel-front:v1.9.2-v4.0.4
  - docker.io/deis/kube-svc-redirect:v1.0.2
  - mcr.microsoft.com/k8s/flexvolume/keyvault-flexvolume:v0.0.13
  - mcr.microsoft.com/k8s/flexvolume/blobfuse-flexvolume:1.0.8
  - gcr.io/google-containers/ip-masq-agent-amd64:v2.5.0
  - k8s.gcr.io/ip-masq-agent-amd64:v2.5.0
  - gcr.io/google-containers/ip-masq-agent-amd64:v2.3.0
  - k8s.gcr.io/ip-masq-agent-amd64:v2.3.0
  - gcr.io/google-containers/ip-masq-agent-amd64:v2.0.0
  - k8s.gcr.io/ip-masq-agent-amd64:v2.0.0
  - nginx:1.13.12-alpine
  - mcr.microsoft.com/k8s/kms/keyvault:v0.0.9
  - quay.io/coreos/flannel:v0.10.0-amd64
  - quay.io/coreos/flannel:v0.8.0-amd64
  - busybox
  - k8s.gcr.io/kube-apiserver:v1.17.0
  - k8s.gcr.io/kube-controller-manager:v1.17.0
  - k8s.gcr.io/kube-proxy:v1.17.0
  - k8s.gcr.io/kube-scheduler:v1.17.0
  - k8s.gcr.io/hyperkube-amd64:v1.16.4
  - k8s.gcr.io/hyperkube-amd64:v1.16.2
  - k8s.gcr.io/hyperkube-amd64:v1.16.1
  - mcr.microsoft.com/k8s/azurestack/core/hyperkube-amd64:v1.16.1-azs
  - k8s.gcr.io/hyperkube-amd64:v1.15.7
  - k8s.gcr.io/cloud-controller-manager-amd64:v1.15.7
  - k8s.gcr.io/hyperkube-amd64:v1.15.5
  - k8s.gcr.io/cloud-controller-manager-amd64:v1.15.5
  - k8s.gcr.io/hyperkube-amd64:v1.15.4
  - k8s.gcr.io/cloud-controller-manager-amd64:v1.15.4
  - mcr.microsoft.com/k8s/azurestack/core/hyperkube-amd64:v1.15.4-azs
  - k8s.gcr.io/hyperkube-amd64:v1.14.8
  - k8s.gcr.io/cloud-controller-manager-amd64:v1.14.8
  - k8s.gcr.io/hyperkube-amd64:v1.14.7
  - k8s.gcr.io/cloud-controller-manager-amd64:v1.14.7
  - mcr.microsoft.com/k8s/azurestack/core/hyperkube-amd64:v1.14.7-azs
  - k8s.gcr.io/hyperkube-amd64:v1.13.12
  - k8s.gcr.io/cloud-controller-manager-amd64:v1.13.12
  - k8s.gcr.io/hyperkube-amd64:v1.13.11
  - k8s.gcr.io/cloud-controller-manager-amd64:v1.13.11
  - mcr.microsoft.com/k8s/core/azure-cloud-controller-manager:v0.3.0
  - mcr.microsoft.com/k8s/core/azure-cloud-node-manager:v0.3.0
  - mcr.microsoft.com/k8s/csi/azuredisk-csi:v0.4.0
  - mcr.microsoft.com/k8s/csi/azurefile-csi:v0.3.0
  - quay.io/k8scsi/csi-attacher:v1.0.1
  - quay.io/k8scsi/csi-cluster-driver-registrar:v1.0.1
  - quay.io/k8scsi/csi-node-driver-registrar:v1.1.0
  - quay.io/k8scsi/csi-provisioner:v1.0.1
  - quay.io/k8scsi/livenessprobe:v1.1.0
  - k8s.gcr.io/node-problem-detector:v0.8.0
  - registry:2.7.1
Using kernel:
Linux version 5.0.0-1027-azure (buildd@lgw01-amd64-033) (gcc version 7.4.0 (Ubuntu 7.4.0-1ubuntu1~18.04.1)) #29~18.04.1-Ubuntu SMP Mon Nov 25 21:18:57 UTC 2019
Install completed successfully on  Thu Dec 12 00:47:26 UTC 2019
VSTS Build NUMBER: 20191211.10
VSTS Build ID: 27331958
Commit: 040ee0bdcaa712eb492e364040ae399035fcfd7a
Feature flags:

**Comment by HeShaobo1** [comment]
The NSG rules don't have limitations, we confirmed that inbound rules and outbound rules. I supposed that due to the **Standard Public SKU** LB, the creating Instance won't be as a backend instance, so it was not able to connect to the Internet.
So under the above architecture, does the instance not have access to the Internet during creation? or it is an issue of 0.45.0 ?

**Comment by CecileRobertMichon** [comment]
Thanks for the details @HeShaobo1. I investigated a bit and looks like this is an existing issue https://github.com/Azure/aks-engine/issues/1970. I have added the issue to the backlog and will advocate with the team that it gets treated as priority.

**Comment by jackfrancis** [comment]
I can definitely confirm that SLB + VMSS + cluster-autoscaler are fully functional, and regularly tested.

For example, we test this cluster configuration against every PR, for versions Kubernetes 1.13 through 1.18:

https://github.com/Azure/aks-engine/blob/master/examples/e2e-tests/kubernetes/release/default/definition.json

Those E2E tests include a cluster-autoscaler scale up and down for every run. These tests gate any merge into master, so they are regularly, 100% green.

We will work against the other configuration properties in the api model provided to figure out what combination of configurations are responsible for scale failures.

**Comment by mboersma** [comment]
I provisioned a cluster using aks-engine v0.45.0 and the apimodel that @vossss posted initially, except in my own VNET and in eastus. I've been scaling in a loop using `aks-engine scale` and manually in the portal, but so far haven't caused any errors.

**Comment by jackfrancis** [comment]
Validated that VMSS + SLB + private cluster scales up fine.

**Comment by jackfrancis** [comment]
Update: I am also not able to reproduce any scaling failures for SLB + VMSS + cluster-autoscaler on a new cluster built in chinanorth2

**Comment by jackfrancis** [comment]
Same thing w/ a private cluster.

**Comment by jackfrancis** [comment]
I've filed #2549 as the result of determining that Standard LB + private cluster configurations don't work at present. We will need to address that in order to unblock cluster scenarios using the api model in this issue.

**Comment by jackfrancis** [comment]
Fixed in #2572
